WebDevStudios/wd_s
# Feature/#456 gutenberg theme setup

Closes #456

### DESCRIPTION ###
Adds various Gutenberg supports to give wd_s basic Gutenberg support out of the box.

Aside from the screenshots below, this also adds `style-editor` support. I didn't wind up adding dark mode support as we can add that on a case-by-case basis.

### SCREENSHOTS ###
## Color Palette:
![](https://dl.dropbox.com/s/o9x84mbkh9inrx4/Screenshot%202019-01-25%2014.06.47.jpg?dl=0)
![](https://dl.dropbox.com/s/sg2zvqlqf83zgen/Screenshot%202019-01-25%2014.20.35.jpg?dl=0)

## Font sizes:
![](https://dl.dropbox.com/s/3y13e2o5uwifby0/Screenshot%202019-01-25%2014.06.59.jpg?dl=0)

## Full-Width Option:
![](https://dl.dropbox.com/s/s777ocju0iu69cq/Screenshot%202019-01-25%2014.07.13.jpg?dl=0)

## Responsive Embeds:
### Without:
![](https://dl.dropbox.com/s/oyyqno3b2n45bhb/Screenshot%202019-01-25%2014.10.24.jpg?dl=0)
![](https://dl.dropbox.com/s/n04xyfmoqdgxil3/Screenshot%202019-01-25%2014.10.04.jpg?dl=0)

### With:
![](https://dl.dropbox.com/s/uitgr77ls53dzvv/Screenshot%202019-01-25%2014.09.24.jpg?dl=0)
![](https://dl.dropbox.com/s/ak4d0jtwdeagsad/Screenshot%202019-01-25%2014.09.42.jpg?dl=0)

Since the video expands to its container width at any screen size, I don't think we should add any size restrictions to it in our styles and instead adjust as needed as we build themes out. If we find that we're consistently setting a `max-width` on the responsive embed class, we can bake something into wd_s.

CBT: https://app.crossbrowsertesting.com/public/i29947718af4369d/screenshots/z64198d684be05c1c8f4

iOS:
![](https://dl.dropbox.com/s/0uyys882tmulw0b/Screenshot%202019-01-25%2014.30.28.jpg?dl=0)
![](https://dl.dropbox.com/s/j1fsiypyjs9yszv/Screenshot%202019-01-25%2014.30.57.jpg?dl=0)

Screenshots in the notes above are from Chrome on Mac OS which, for whatever reason, got all weird in CBT.

### OTHER ###

### STEPS TO VERIFY ###
1. Make sure your local is up to date with the latest and greatest
2. Add a Gutenberg text block and set the background and text colors (note that they use the colors set in the `/inc/acf.php` file)
3. Adjust the text of your block using the Gutenberg font size settings
4. Add an image block and set it to full-width
5. Add a YouTube video via oEmbed
6. Check it all out on the front end to see the colors, size, and responsive embeds in action

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

Could probably add some documentation around the default settings, like font size, and where to find the `style-editor.css` file to edit.
